p2p/src/authenticated/actors/tracker/reservation.rs

```diff
@@ -39,26 +38,22 @@ impl<E: Spawner + Metrics, P: PublicKey> Reservation<E, P> {
 
 impl<E: Spawner + Metrics, P: PublicKey> Drop for Reservation<E, P> {
     fn drop(&mut self) {
-        let mut closer = self.closer.take().expect("Reservation::drop called twice");
+        let mut releaser = self
+            .releaser
+            .take()
+            .expect("Reservation::drop called twice");
 
         // If the mailbox is not full, we can release the reservation immediately without spawning a task.
-        let Err(e) = closer.try_send(self.metadata.clone()) else {
+        if releaser.try_release(self.metadata.clone()) {
             // Sent successfully, nothing to do.
             return;
         };
-        if e.is_full() {
-            // If the mailbox is full, we need to spawn a task to handle the release. If we used `block_on` here,
-            // it could cause a deadlock.
-            let metadata = self.metadata.clone();
-            self.context.spawn_ref()(async move {
-                closer.send(metadata).await.unwrap();
-            });
-        } else {
-            // If any other error occurs, we should panic!
-            panic!(
-                "unexpected error while trying to release reservation: {:?}",
-                e
-            );
-        }
+
+        // If the mailbox is full, we need to spawn a task to handle the release. If we used `block_on` here,
```

Comment:
Would we actually deadlock here if we were to wait until we can send the release message? I think that as long as the tracker can still make progress (i.e. drain items from its `receiver`) then `drop` should eventually return. Is there a case where the tracker can't make progress until `drop` completed?